src/Microsoft.VisualStudio.ProjectSystem.Managed/Microsoft.VisualStudio.ProjectSystem.Managed.csproj

```diff
@@ -28,154 +28,64 @@
     <PackageReference Include="Microsoft.VisualStudio.RpcContracts" />
   </ItemGroup>
 
+  <ItemGroup>
+    <!-- Add 'DependentUpon' metadata based on file names. This causes .cs files to nest beneath .xaml files in Solution Explorer. -->
+    <Compile Update="ProjectSystem\Rules\**\*.cs">
+      <DependentUpon>%(Filename).xaml</DependentUpon>
+    </Compile>
+  </ItemGroup>
+
+  <!--
+    NOTE ON XAML RULES
+
+    We use one of two item types for XAML rule files:
+
+    - "XamlPropertyRule" for rules that require a generated class. Useful when working with Dataflow project rule sources.
+    - "XamlPropertyRuleNoCodeBehind" for rules that do not require a generated class.
+
+    In addition to the above, there are two metadata that control how rules are treated during build:
+
+    - "XlfInput" controls whether display strings in the rule file should be localised. Defaults to true.
+    - "IsEmbedded" indicates that the rule is embedded (see RuleExporter.cs).
+    - "GenerateProjectPropertiesClass" indicates that code generation should contribute to the partial "ProjectProperties" class.
```

Comment:
It would be useful to list the defaults for "IsEmbedded" and "GenerateProjectPropertiesClass".